oak-jackrabbit-api/src/main/java/org/apache/jackrabbit/api/JackrabbitSession.java

```diff
@@ -23,15 +23,21 @@
 import javax.jcr.Node;
 import javax.jcr.Property;
 import javax.jcr.Session;
+
+import java.security.Principal;
+import java.util.Set;
+
 import javax.jcr.AccessDeniedException;
 import javax.jcr.RepositoryException;
 import javax.jcr.UnsupportedRepositoryOperationException;
 
 import org.jetbrains.annotations.NotNull;
+import org.osgi.annotation.versioning.ProviderType;
 
 /**
  * Jackrabbit specific extension of the JCR {@link javax.jcr.Session} interface.
  */
+@ProviderType
```

Comment:
This was not marked as provider type before. The default is consumer type which requires a major version change for each added method.